sympy/solvers/solveset.py

```diff
@@ -455,6 +455,9 @@ def solveset_real(f, symbol):
     original_eq = f
     f = together(f)
 
+    if not symbol in fraction(f)[1].free_symbols and f.is_rational_function():
```

Comment:
If you look at [these](https://github.com/sympy/sympy/blob/master/sympy/solvers/solveset.py#L777-L780) similar lines in `solveset_complex`. These exact line would not work in `solveset_real`, but they have to be added with `if not symbol in fraction(f)[1].free_symbols` condition.
> Does this cause any problems?

Yes, it would run into an infinite loop.